grassroots-frontend/src/routes/create-contact.tsx

```diff
@@ -1,10 +1,66 @@
 import { createFileRoute } from "@tanstack/react-router";
-import { JSX } from "react";
+import { JSX, useCallback } from "react";
+
+import { TextInput } from "@mantine/core";
+import { CreateContactInDto } from "../grassroots-shared/contact.entity.dto";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { grassrootsAPI } from "../grassRootsAPI";
 
 export const Route = createFileRoute("/create-contact")({
-  component: RouteComponent,
+  component: CreateContact,
 });
 
-function RouteComponent(): JSX.Element {
-  return <div>TODO &quot;/create-contact&quot;!</div>;
+function CreateContact(): JSX.Element {
+  // TODO: enable input validation.
+  const form =
+    useForm<CreateContactInDto>(/*{
+    resolver: classValidatorResolver(PendingContactInDto),
+    mode: "onBlur",
+  }*/);
+
+  const queryClient = useQueryClient();
+  const { mutateAsync } = useMutation({
+    mutationFn: async (contact: CreateContactInDto) => {
+      const result = await grassrootsAPI.POST("/contacts", {
+        body: contact,
+      });
+      if (!result.data) {
+        throw new Error("Failed to create contact.");
+      }
+      return result.data;
+    },
+    retry: 1,
+    onSuccess: async () => {
+      await queryClient.invalidateQueries({ queryKey: ["contacts"] });
+    },
+  });
+
+  const onSubmit: SubmitHandler<CreateContactInDto> = useCallback(
+    async (data) => {
+      await mutateAsync(data);
+      form.reset();
+    },
+    [],
+  );
+
+  return (
+    <FormProvider {...form}>
+      {/* This little typescript dance is required to make eslint happy.  */}
+      <form onSubmit={(...args) => void form.handleSubmit(onSubmit)(...args)}>
+        <TextInput
+          id="firstName"
```

Comment:
Why is ID needed on this element but not others?

Reply:
It's not, removing.